### PR TITLE
fix(NEWRELIC-7409): allow ESM modules to be defined via package.json

### PR DESCRIPTION
**Pull request by jordigh**
This checks to see if the file was found but attempted require of a module, so we attempt an import instead.

### CODE REVIEW

File: nodejs/index.js [hunk 1]
```diff
@@ -42,13 +42,23 @@ function handleRequireImportError(e, moduleToImport) {
...
     // require failed, it could be an es module, so we try to import as mjs
     if (e.code === 'MODULE_NOT_FOUND') {
       try {
-        return await import(`${LAMBDA_TASK_ROOT}/${moduleToImport}.mjs`)
+        return await import(`${modpath}.mjs`)
+      } catch (esmError) {
+        throw handleRequireImportError(esmError, moduleToImport)
+      }
+    } else if (e.code === 'ERR_REQUIRE_ESM') {
+      // The name is right, but we attempted to require an ESM module,
```
Comment (mrickard) [inline nodejs/index.js:57]:
```suggestion
      // The name is right, but we attempted to require an ECMAScript module,
```

Comment (jordigh) [reply nodejs/index.js]: Right after I said that I can't stop saying "ESM module"...

**Review by mrickard — CHANGES_REQUESTED** [review]
This will test locally, but there's also a group of ciTests--the CI-run tests don't have access to `time` and `tap`.

**Comment by jordigh** [comment]
Okay, does that look good?

**Review by mrickard — APPROVED** [review]

**Pull request merged**